src/main/java/com/iotiq/application/service/ImageService.java

```diff
@@ -0,0 +1,31 @@
+package com.iotiq.application.service;
+
+import com.iotiq.application.messages.image.ImageDto;
+import org.springframework.stereotype.Service;
+import org.springframework.web.multipart.MultipartFile;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.StandardCopyOption;
+import java.util.UUID;
+
+@Service
+public class ImageService {
+
+    // Save image in a local directory
+    public ImageDto saveImageToStorage(String uploadDirectory, MultipartFile imageFile) throws IOException {
+        String uniqueFileName = UUID.randomUUID().toString() + "_" + imageFile.getOriginalFilename();
```

Comment:
Suggestion: may be, we sanitize the file name:
String originalFileName = StringUtils.cleanPath(imageFile.getOriginalFilename());
if (originalFileName.contains("..")) {
    throw new IOException("Invalid file path");
}